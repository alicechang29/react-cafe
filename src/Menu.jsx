import { Link } from "react-router-dom";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem,
} from "reactstrap";
import "./Menu.css";

/** Render list of food or drink items.
 *
 * Params:
 * - items: [{id, name, description, recipe, serve}, ...]
 */

function Menu({ items, type }) {
  console.log("* Menu %o", items, type);
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="fw-bold text-center">
            {type.toUpperCase()} Menu
          </CardTitle>
          <CardText>
            Try one of our delicious menu items!
          </CardText>
          <ListGroup>
            {items.map(({ id, name }) => (
              <Link to={`/${type}/${id}`} key={id}>
                <ListGroupItem>{name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
