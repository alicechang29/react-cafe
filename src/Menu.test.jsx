import { describe, it, expect } from "vitest";

import { screen, fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Menu from "./Menu";

it("renders without crashing for snacks", async function () {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/snacks"]}>
      <Menu type="snacks" items={[{ id: 0, name: "test snack" }]} />
    </MemoryRouter>
  );
  screen.getByText("test snack");
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing for drinks", async function () {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/drinks"]}>
      <Menu type="drinks" items={[{ id: 0, name: "test drink" }]} />
    </MemoryRouter>
  );
  screen.getByText("test drink");

  expect(asFragment()).toMatchSnapshot();
});
