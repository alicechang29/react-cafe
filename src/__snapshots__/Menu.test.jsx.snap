// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing for drinks 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="fw-bold text-center card-title"
        >
          DRINKS Menu
        </div>
        <p
          class="card-text"
        >
          Try one of our delicious menu items!
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/drinks/0"
          >
            <li
              class="list-group-item"
            >
              test drink
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders without crashing for snacks 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="fw-bold text-center card-title"
        >
          SNACKS Menu
        </div>
        <p
          class="card-text"
        >
          Try one of our delicious menu items!
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/0"
          >
            <li
              class="list-group-item"
            >
              test snack
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
